examples/core/transform-feedback-instanced/app.js

```diff
@@ -0,0 +1,348 @@
+    out float v_rotation;
+
+    float rand(vec2 co)
+    {
+        return fract(sin(dot(co.xy ,vec2(12.9898,78.233))) * 43758.5453);
+    }
+
+    void main()
+    {
+        float theta = M_2PI * rand(vec2(u_time, a_rotation + a_offset.x + a_offset.y));
+
+        float cos_r = cos(a_rotation);
+        float sin_r = sin(a_rotation);
+        mat2 rot = mat2(
+            cos_r, sin_r,
+            -sin_r, cos_r
+        );
+
+        vec2 p = WANDER_CIRCLE_R * vec2(cos(theta), sin(theta)) + vec2(WANDER_CIRCLE_OFFSET, 0.0);
+        vec2 move = normalize(rot * p);
+        v_rotation = atan(move.y, move.x);
+
+        v_offset = a_offset + MOVE_DELTA * move;
+
+        // wrapping at edges
+        v_offset = vec2 (
+            v_offset.x > MAP_HALF_LENGTH ? - MAP_HALF_LENGTH : ( v_offset.x < - MAP_HALF_LENGTH ? MAP_HALF_LENGTH : v_offset.x ) ,
+            v_offset.y > MAP_HALF_LENGTH ? - MAP_HALF_LENGTH : ( v_offset.y < - MAP_HALF_LENGTH ? MAP_HALF_LENGTH : v_offset.y )
+            );
+
+        gl_Position = vec4(v_offset, 0.0, 1.0);
+    }
+`;
+
+const FS_EMIT = `#version 300 es
+  precision highp float;
+  precision highp int;
+
+  void main()
+  {
+  }
+`;
+
+const VS_DRAW = `#version 300 es
+  #define OFFSET_LOCATION 0
+  #define ROTATION_LOCATION 1
+  #define POSITION_LOCATION 2
+  #define COLOR_LOCATION 3
+
+  precision highp float;
+  precision highp int;
+
+  layout(location = POSITION_LOCATION) in vec2 a_position;
+  layout(location = ROTATION_LOCATION) in float a_rotation;
+  layout(location = OFFSET_LOCATION) in vec2 a_offset;
+  layout(location = COLOR_LOCATION) in vec3 a_color;
+
+  out vec3 v_color;
+
+  void main()
+  {
+      v_color = a_color;
+
+      float cos_r = cos(a_rotation);
+      float sin_r = sin(a_rotation);
+      mat2 rot = mat2(
+          cos_r, sin_r,
+          -sin_r, cos_r
+      );
+      gl_Position = vec4(rot * a_position + a_offset, 0.0, 1.0);
+
+      // -hack-
+      // gl_Position = vec4(a_position, 0.0, 1.0);
+      // v_color.xy = a_offset;
+      // v_color.z = a_rotation;
+}
+`;
+
+const FS_DRAW = `#version 300 es
+  #define ALPHA 0.9
+
+  precision highp float;
+  precision highp int;
+
+  in vec3 v_color;
+
+  out vec4 color;
+
+  void main()
+  {
+      color = vec4(v_color * ALPHA, ALPHA);
+}
+`;
+
+const NUM_INSTANCES = 1000;
+let currentSourceIdx = 0;
+
+const animationLoop = new AnimationLoop({
+  glOptions: {webgl2: true},
+  // eslint-disable-next-line
+  onInitialize({canvas, gl}) {
+
+    const modelRender = new Model(gl, {
+      id: 'Model-Render',
+      vs: VS_DRAW,
+      fs: FS_DRAW,
+      geometry: new Geometry({
+        id: 'Model-Render-Geometry',
+        drawMode: gl.TRIANGLE_FAN,
+        vertexCount: 3
+      }),
+      isInstanced: true,
+      instanceCount: NUM_INSTANCES
+    });
+
+    const modelTransform = new Model(gl, {
+      id: 'Model-Transform',
+      vs: VS_EMIT,
+      fs: FS_EMIT,
+      varyings: VARYINGS,
+      geometry: new Geometry({
```

Comment:
A Geometry without any "geometry". Doesn't make much sense.

We should consider how `VertexArray's impact the API, I see you have a ticket open, will comment there.